### PR TITLE
Avoid generating unnecessary constraints for lambdas and deplambdas

### PR DESCRIPTION
**Pull request by umazalakain**
Minor change that avoids generating constraints on fresh identifiers that are not mentioned anywhere else

### CODE REVIEW

File: src/main/scala/rise/core/DSL/infer.scala [hunk 1]
```diff
@@ -136,9 +139,8 @@ object infer {
       val env1 : Map[String, Type] = env + (tx.name -> tx.t)
       val (te, cs, ftvE) = constrainTypes(env1)(e)
       val ft = FunType(tx.t, te.t)
-      val exprT = genType(expr)
-      val c = TypeConstraint(exprT, ft)
-      (Lambda(tx, te)(ft), cs :+ c, ftvE)
+      val cs1 = constrIfTyped(expr.t)(TypeConstraint(expr.t, ft))
+      (Lambda(tx, te)(ft), cs :++ cs1, ftvE)
```
Comment (Bastacyclop) [inline src/main/scala/rise/core/DSL/infer.scala:143]: minor style: we usually use `++` instead of `:++`, I think that they do the same.

**Review by Bastacyclop — COMMENTED** [review]
@umazalakain what is the motivation for this change? does it make inference faster?

**Comment by umazalakain** [comment]
@Bastacyclop It avoids adding (and solving) constraints that do not constraint type inference in any way whatsoever. The tests run in 4min5sec on this branch versus 4min49sec on master, though one of the `mmTensorCheck` tests fails on here, so it might be slightly misleading.

File: src/main/scala/rise/core/DSL/infer.scala (hunk 1 above)
Comment (umazalakain) [reply src/main/scala/rise/core/DSL/infer.scala]: Fixed

**Review by michel-steuwer — APPROVED** [review]
Fine with me. @umazalakain you will have to resolve a merge conflict first.

**Review by michel-steuwer — APPROVED** [review]
👍

**Pull request merged**